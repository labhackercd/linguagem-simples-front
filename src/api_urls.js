module.exports = Object.freeze({

    //NAME CONSTANTS
    APPLICATION_NAME: "Estudio",

    //APPLICATION URL'S
        APPLICATION_FRONT_BASE_URL: "www.teste.com",  //APPLICATION FRONT BASE URL
        //APPLICATION_SERVER_API_BASE_URL: 'http://localhost:8000/api/', //APPLICATION API SERVER BASE URL
        APPLICATION_SERVER_API_BASE_URL: 'https://acompanhe.camara.leg.br/server/api/',
        APPLICATION_RESET_PASSWORD_URL: 'https://acompanhe.camara.leg.br/server/password_reset/',

    //ROUTES
        INITIAL_PAGE_URL: '/', //Application Login and Initial page
        ESTUDIO_PAGE_URL: '/estudio', //Application Estudio Page
        DASHBOARD_PAGE_URL: '/dashboard/:dashboardId', //Application Dashboard page
        DASHBOARD_BASE_URL: '/dashboard/',

    //API'S CAMARA DOS DEPUTADOS
        API_CD_EVENTOS_PLENARIO: 'https://dadosabertos.camara.leg.br/api/v2/eventos?codTipoEvento=110',// 110 code corresponds to Plenary
        API_CD_EVENTOS: 'https://dadosabertos.camara.leg.br/api/v2/eventos/',
        API_CD_ORIENTACAO_VOTACAO: 'https://liderancadigital.camara.leg.br/proxy_pass/secod-wsVotDecom-orientacao',
        API_CD_ITENS_EM_VOTACAO: 'https://infoleg.camara.gov.br/wsVotDecom/votacao/itens-em-votacao-na-reuniao/',

    //API'S BACKEND
        //AUTH URL'S
            TOKEN_REFRESH_URL: '/token/refresh/', //Application Api Token REFRESH PATH
            TOKEN_OBTAIN_URL: '/token/obtain/', //Application Api Token PATH
            TOKEN_VERIFY_URL: '/token/verify/', //Application Api Token Verify

        //FILE-VIDEO
            API_FILE_VIDEO_URL:'/file-video/', //Application end-point to obtain the source of specific video

        //NEWS
            API_NEWS_URL:'/newsâ€‹/news_list', //Obtain news list
            API_NEWS_SEARCH_URL:'/news-search/', //Obtain specific news using parameters

        //PUBLICATIONS
            API_PUBLICATIONS_URL:'/publications/',

        //RADIO AGENCY
            API_RADIO_AGENCY_URL:'/radioagency/',
            API_RADIO_AGENCY_SEARCH_URL:'/radioagency-search/',

        //RADIO CAMARA
            API_RADIO_CAMARA_URL:'/radiocamara/',
            API_RADIO_CAMARA_SEARCH_URL:'/radiocamara-search/',

        //SAVED CONTENTS
            API_SAVED_CONTENTS_URL:'/saved-contents/',

        //SESSIONS
            API_SESSIONS_URL:'/sessions/',

        //TV CAMARA
            API_TV_CAMARA_URL:'/tvcamara/',
            API_TV_CAMARA_SEARCH_URL:'/tvcamara-search/',

        //VIDEOS
            API_VIDEOS_SNIPPETS:'/videos-session/'
  });
