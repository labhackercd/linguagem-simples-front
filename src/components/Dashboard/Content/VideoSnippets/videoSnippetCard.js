import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import ButtonBase from '@material-ui/core/ButtonBase'
import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'

export default function SnippetCard(props){
  const deputado = props.data.author;
  const siglaPartidaria = props.data.legend;

  const [open, setOpen] = React.useState(false);


  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = (value) => {
    setOpen(false);
  };

  return (
    <Box width="100%" height="100%" marginX={0.3} marginTop={0.5}>
      <ButtonBase onClick={handleClickOpen} id={"openModal"}>
        <div>
            <img style={{width:"100%", height:"6vh"}} id="image" src={props.data.thumbnail} alt="Thumbanail trecho"></img>
            <Box fontSize={10}>
            <Typography>{deputado}</Typography>
            </Box>
            <Box fontSize={8}>
            <Typography variant="body2">{siglaPartidaria}</Typography>
            </Box>
        </div>
        </ButtonBase>
      <VideoSnippetModal open={open} onClose={handleClose} data={props.data} />
    </Box>
  );
}