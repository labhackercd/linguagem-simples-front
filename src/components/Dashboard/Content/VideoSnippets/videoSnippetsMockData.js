 export const videoSnippetsMockData = [
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234414000",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h53'34\"\nduração - 00:00:58",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234452950",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h54'12\"\nduração - 00:00:59",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234492497",
    "description": "Paulão\n                                                                        \nDeputado PT-AL\n\nhorário - 15h54'52\"\nduração - 00:02:33",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/171617.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234625877",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h57'05\"\nduração - 00:00:34",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
    "description": "Arthur Oliveira Maia\n                                                                        \nDeputado DEM-BA\n\nhorário - 15h57'20\"\nduração - 00:02:04",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234744800",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 15h59'04\"\nduração - 00:01:56",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234841110",
    "description": "Alessandro Molon\n                                                                        \nDeputado PSB-RJ\n\nhorário - 16h00'41\"\nduração - 00:00:52",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160511.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234874013",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h01'14\"\nduração - 00:01:14",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234928823",
    "description": "Paulão\n                                                                        \nDeputado PT-AL\n\nhorário - 16h02'08\"\nduração - 00:01:02",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/171617.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234971410",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h02'51\"\nduração - 00:00:34",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234986130",
    "description": "Fernanda Melchionna\n                                                                        \nDeputado PSOL-RS\n\nhorário - 16h03'06\"\nduração - 00:01:08",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/204407.jpg"
  },
  {
    "id": 12,
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594235034203",
    "description": "Rodrigo Maia\n                                                                        \nDeputado DEM-RJ\n\nhorário - 16h03'54\"\nduração - 00:01:21",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/74693.jpg"
  },
  {
    "id": 13,
    "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594235095990",
    "description": "Arnaldo Jardim\n                                                                        \nDeputado CIDADANIA-SP\n\nhorário - 16h04'55\"\nduração - 00:00:22",
    "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/141391.jpg"
  }
]