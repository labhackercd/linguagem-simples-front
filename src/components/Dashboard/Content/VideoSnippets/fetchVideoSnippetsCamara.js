import axiosInstance from '../../../../auth/axiosApi'
import {API_VIDEOS_SNIPPETS} from '../../../../api_urls'


export default async function fetchSessionVideos(sessionIdDadosAbertos){
    const response = await axiosInstance.get((API_VIDEOS_SNIPPETS+sessionIdDadosAbertos), {});

    if(response.status===200){
        return response.data;
    }else{
        return null;
    }
}
