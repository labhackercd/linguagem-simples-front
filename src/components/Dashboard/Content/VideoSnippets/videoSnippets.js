import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import CircularProgress from '@material-ui/core/CircularProgress';
import {videoSnippetsMockData} from './videoSnippetsMockData'
import fetchSessionVideos from './fetchVideoSnippetsCamara'
import { FixedSizeList } from 'react-window';
import ListItem from '@material-ui/core/ListItem';
import InputAdornment from '@material-ui/core/InputAdornment';
import SearchIcon from '@material-ui/icons/Search';
import TextField from '@material-ui/core/TextField';
import ButtonBase from '@material-ui/core/ButtonBase'

import VideoSnippetModal from './VideoSnippetModal/videoSnippetModal'
import SnippetCard from './videoSnippetCard'



export default class VideoSnippets extends React.Component {
    
  constructor(props){
    super(props);
    this.state = { 
        snippets: '',
        filteredSnippets: '',
        sessionHasStarted: false,
        dataLoaded: false,
        searchField: '',
        error:{
          happened:false,
          message: "Erro",
          status:""
        }
    };
  }

  componentDidMount(){
      this._isMounted = true;
      //console.log(this.props)
      if(this._isMounted){

          if(this.props.sessionInfo.situation_session === "pre_session"){
            this.setState({sessionHasStarted:false});
            //this.setState({snippets:true});
          }else{
            this.setState({sessionHasStarted:true})
            this.fetchSessionsList();
          }
          
      }
  }

  fetchSessionsList = async term => {
    try {
      //console.log(this.props)
      const data = await fetchSessionVideos(this.props.sessionInfo.id_session_dados_abertos);
      this.setState({snippets:data})
      this.setState({filteredSnippets:data})
      this.setState({dataLoaded:true});
    } catch (error) {
      //console.log(error)
      this.setState({error:{happened:true,status:error.response.status} })
    }
  };

  renderSearchBarFunction(){
    return(
      <React.Fragment>
        <Grid container>
          <Grid item xs={10}>
            <Box marginLeft={2} marginTop={1}>
              <Typography variant="h5" style={{ color: "#3E3E3E" }}>Trechos</Typography>
            </Box>
          </Grid>
          <Grid item xs={2}>
            <Box marginRight={1}> 
              <TextField
                id="input-search-snippet"
                size="small"
                onChange={this.snippetsFilterOnChange}
                InputProps={{
                  endAdornment: (<InputAdornment position="start"><SearchIcon /></InputAdornment>),
                }}/>
            </Box>
          </Grid>
        </Grid>
      </React.Fragment>
    )
  }

  renderListItem = ({index, style}) => {
    return(
      <ListItem disableGutters={true} style={style} >
          <SnippetCard data = {this.state.filteredSnippets[index]}></SnippetCard>        
      </ListItem>
    )
  }

  snippetsFilterOnChange = (event) => {
    //console.log(event.target.value)
    this.setState({
      searchField: event.target.value
    })
    
    this.setState({
      filteredSnippets: this.state.snippets.filter(term => term.author.toLowerCase().includes(this.state.searchField.toLowerCase()))
    })
    
  }

  render(){
    const widthSnippetsBox = parseInt(((window.innerWidth)*0.54));
    const widthSnippetsItem = (((window.innerWidth)*0.5)*0.12);
    //console.log(widthSnippetsBox)
   
    if(this.state.error.happened){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>Erro</Typography>
              </Box>)
    }

    if(!this.state.sessionHasStarted){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}>
                  <Typography>A transmissão ainda não foi iniciada.</Typography>
              </Box>)
    }

    if(!this.state.dataLoaded){
      return (<Box display="flex" justifyContent="center" alignItems="center" width={"100%"}><CircularProgress></CircularProgress></Box>)
    }

    return (
      <div>
        <Grid container >
        {this.renderSearchBarFunction()}
          <Grid item xs={12}>
            <Box marginLeft={1}>
              <FixedSizeList width={widthSnippetsBox} height={"12.5vh"} itemSize={widthSnippetsItem} layout="horizontal" itemCount={this.state.filteredSnippets.length}>        
                {this.renderListItem}
              </FixedSizeList>
            </Box>
          </Grid>
        </Grid>
      </div>
    )
  }
}