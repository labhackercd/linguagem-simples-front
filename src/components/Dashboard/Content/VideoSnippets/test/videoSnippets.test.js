import React from 'react';
import { render, screen } from '@testing-library/react';
import {shallow} from "enzyme/build";
import VideoSnippets from '../videoSnippets'
import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import CircularProgress from '@material-ui/core/CircularProgress';
import SnippetCard from '../videoSnippetCard'
import {API_VIDEOS_SNIPPETS} from '../../../../../api_urls'
import axiosInstance from '../../../../../auth/axiosApi'

  
describe('Testing VideoSnippet Component', () => {
    const mockSessionId = 1;
    const mockSessionInfoPreSessionStatus ={
        "id": 1,
        "author": {
          "id": 1,
          "is_superuser": true,
          "username": "jpnsoares",
          "first_name": "",
          "last_name": "",
          "email": "jpnsoares@email.com",
          "profile": "editor"
        },
        "location": "plenary",
        "date": "2020-09-01",
        "type_session": "virtual",
        "situation_session": "pre_session",
        "resume": "",
        "enable": true,
        "id_session_dados_abertos": "59895"
    }
    const mockSessionInfoStartedSessionStatus ={
        "id": 1,
        "author": {
          "id": 1,
          "is_superuser": true,
          "username": "jpnsoares",
          "first_name": "",
          "last_name": "",
          "email": "jpnsoares@email.com",
          "profile": "editor"
        },
        "location": "plenary",
        "date": "2020-09-01",
        "type_session": "virtual",
        "situation_session": "started_session",
        "resume": "",
        "enable": true,
        "id_session_dados_abertos": "59895"
    }

    test("Should render the VideoSnippets section and match snapshot", async (done) => {
        const component = mount(<VideoSnippets sessionId={mockSessionId} sessionInfo={mockSessionInfoPreSessionStatus}></VideoSnippets>);

        expect(component.exists()).toEqual(true);
        expect(component).toMatchSnapshot();
        done();
    });
   
    test("Should render pre session div information when session status is PRE_SESSION", async (done) => {
        const component = mount(<VideoSnippets sessionId={mockSessionId} sessionInfo={mockSessionInfoPreSessionStatus}></VideoSnippets>);
        expect(component.text()).toMatch(/A transmissão/i)
        done();
    });
   
    test("Should render CircularProgress when session status is not pre session and data is been loading", async (done) => {
        const component = await mount(<VideoSnippets sessionId={mockSessionId} sessionInfo={mockSessionInfoStartedSessionStatus}></VideoSnippets>);
        setImmediate(() => {
            component.update();
            //console.log(component.debug())
            const containsSpinner = component.containsMatchingElement(<CircularProgress />);
            expect(containsSpinner).toBeTruthy()
            done();
         })
    });
     
    test("Should return error", async (done) => {
        var mock = new MockAdapter(axiosInstance);
        var mockResponse = {
            "data": [
              {
                "url": "https://www.camara.leg.br/evento-legislativo/59895/sessao/523297/video-trecho/1598981385760",
                "author": "Deputado Y",
                "legend": "Deputado PL-RJ",
                "schedule": "horário - 14h29'45",
                "duration": "duração - 00:01:42",
                "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/178946.jpg"
              },
              {
                "url": "https://www.camara.leg.br/evento-legislativo/59895/sessao/523297/video-trecho/1598981381640",
                "author": "Deputado  X",
                "legend": "Deputado PL-RJ",
                "schedule": "horário - 14h29'41",
                "duration": "duração - 00:00:24",
                "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/178946.jpg"
              },
            ],
            "status": 200,
            "statusText": "OK",
            "headers": {
              "content-length": "47681",
              "content-type": "application/json"
            },
            "config": {
              "url": "/videos-session/59895",
              "method": "get",
              "headers": {
                "Accept": "application/json",
                "Authorization": "JWT eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjAwMzcwMjk4LCJqdGkiOiI0ODNiNjA5NDgzMzk0MTcwYjVlY2I3OTBiNTM2ZDI2YiIsInVzZXJfaWQiOjEsInVzZXJuYW1lIjoiYWRtaW4ifQ.oviGcjwER71YKCL0siWx5O00P87qNxKzoT8DA9XJjAQ"
              },
              "baseURL": "https://acompanhe.camara.leg.br/server/api/",
              "transformRequest": [
                null
              ],
              "transformResponse": [
                null
              ],
              "timeout": 0,
              "xsrfCookieName": "XSRF-TOKEN",
              "xsrfHeaderName": "X-XSRF-TOKEN",
              "maxContentLength": -1
            },
            "request": {}
        }

        await mock.onGet(API_VIDEOS_SNIPPETS+mockSessionInfoStartedSessionStatus.id_session_dados_abertos).reply(404,mockResponse);
        const component = await mount(<VideoSnippets sessionId={mockSessionId} sessionInfo={mockSessionInfoStartedSessionStatus}></VideoSnippets>);
      
        setImmediate(() => {
            expect(component.text()).toMatch(/Erro/i)
            mock.restore();
            done();
         })
    });
    
    test("Should render cards when session status is not pre session and data has been loaded", async (done) => {
        var mock = new MockAdapter(axiosInstance);
        var mockResponse = {
            "data": [
              {
                "url": "https://www.camara.leg.br/evento-legislativo/59895/sessao/523297/video-trecho/1598981385760",
                "author": "Deputado Y",
                "legend": "Deputado PL-RJ",
                "schedule": "horário - 14h29'45",
                "duration": "duração - 00:01:42",
                "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/178946.jpg"
              },
              {
                "url": "https://www.camara.leg.br/evento-legislativo/59895/sessao/523297/video-trecho/1598981381640",
                "author": "Deputado  X",
                "legend": "Deputado PL-RJ",
                "schedule": "horário - 14h29'41",
                "duration": "duração - 00:00:24",
                "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/178946.jpg"
              },
            ],
            "status": 200,
            "statusText": "OK",
            "headers": {
              "content-length": "47681",
              "content-type": "application/json"
            },
            "config": {
              "url": "/videos-session/59895",
              "method": "get",
              "headers": {
                "Accept": "application/json",
                "Authorization": "JWT eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNjAwMzcwMjk4LCJqdGkiOiI0ODNiNjA5NDgzMzk0MTcwYjVlY2I3OTBiNTM2ZDI2YiIsInVzZXJfaWQiOjEsInVzZXJuYW1lIjoiYWRtaW4ifQ.oviGcjwER71YKCL0siWx5O00P87qNxKzoT8DA9XJjAQ"
              },
              "baseURL": "https://acompanhe.camara.leg.br/server/api/",
              "transformRequest": [
                null
              ],
              "transformResponse": [
                null
              ],
              "timeout": 0,
              "xsrfCookieName": "XSRF-TOKEN",
              "xsrfHeaderName": "X-XSRF-TOKEN",
              "maxContentLength": -1
            },
            "request": {}
        }

        await mock.onGet(API_VIDEOS_SNIPPETS+mockSessionInfoStartedSessionStatus.id_session_dados_abertos).reply(200,mockResponse);
        const component = await mount(<VideoSnippets sessionId={mockSessionId} sessionInfo={mockSessionInfoStartedSessionStatus}></VideoSnippets>);

        setImmediate(() => {
            component.update();
            //console.log(component.debug())
            const containsSpinner = component.containsMatchingElement(<CircularProgress />);
            expect(containsSpinner).not.toBeTruthy();
            
            //const searchField = component.find("input").at(0);
            
            //console.log(searchField.debug());
            //searchField.instance().value = "x";
            //component.update();
            //searchField.simulate("change");
            mock.restore();
            done();
         })
    });
});

describe('Testing VideoSnippetCard Component', () => {

    const mockedData={
        "url": "https://www.camara.leg.br/evento-legislativo/59733/sessao/523169/video-trecho/1594234640207",
        "author": "Arthur Oliveira Maia",
        "legend": "Deputado DEM-BA",
        "schedule": "horário - 15h57'20",
        "duration": "duração - 00:02:04",
        "thumbnail": "https://www.camara.leg.br/internet/deputado/bandep/pagina_do_deputado/160600.jpg"
    }

    it("should render the Card component and match snapshot ", () => {
        
        const component = shallow(<SnippetCard data={mockedData}/>);

        expect(component.exists()).toEqual(true);
        expect(component).toMatchSnapshot();
        
    });
    
    test("Test Card handleOpen and handleClose", () => {
        
        
        const component =  mount(<SnippetCard data={mockedData}/>);

        //console.log(component.debug())
        const button = component.find("#openModal").at(0);
        //console.log(button.debug())
        button.simulate('click')
        component.update();
        //console.log(component.debug())
        const closeButton = component.find("#closeVideoSnippetModal").at(0)
        closeButton.simulate('click')

    });
    /*
        test("Test Card handleOpen and handleClose", async(done) => {
        var mock = new MockAdapter(axiosInstance);
        
        await mock.onPost('/file-video/').replyOnce(200,"www.video.com");   
        const component = await mount(<SnippetCard data={mockedData}/>);


        setImmediate(() => {
            component.update();
            //console.log(component.debug())
            const button = component.find("#openModal").at(0);
            //console.log(button.debug())
            button.simulate('click')
            component.update();
            //console.log(component.debug())
            const closeButton = component.find("#closeVideoSnippetModal").at(0)
            closeButton.simulate('click')
            mock.restore();
            done();
        });

    });*/
  
  });
  
