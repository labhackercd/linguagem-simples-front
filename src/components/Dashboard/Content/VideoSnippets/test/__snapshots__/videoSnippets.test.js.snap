// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoSnippet Component Should render the VideoSnippets section and match snapshot 1`] = `
<VideoSnippets
  sessionId={1}
  sessionInfo={
    Object {
      "author": Object {
        "email": "jpnsoares@email.com",
        "first_name": "",
        "id": 1,
        "is_superuser": true,
        "last_name": "",
        "profile": "editor",
      },
      "date": "2020-09-01",
      "enable": true,
      "id": 1,
      "id_session_dados_abertos": "59895",
      "location": "plenary",
      "resume": "",
      "situation_session": "pre_session",
      "type_session": "virtual",
    }
  }
>
  <Styled(MuiBox)
    height={1}
    width={1}
  >
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <Styled(MuiBox)
        alignItems="center"
        display="flex"
        height="100%"
        justifyContent="center"
        width="100%"
      >
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <WithStyles(ForwardRef(Typography))
            style={
              Object {
                "color": "#C4C4C4",
              }
            }
            variant="h5"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              style={
                Object {
                  "color": "#C4C4C4",
                }
              }
              variant="h5"
            >
              <h5
                className="MuiTypography-root MuiTypography-h5"
                style={
                  Object {
                    "color": "#C4C4C4",
                  }
                }
              >
                Trechos não disponíveis
              </h5>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </Styled(MuiBox)>
    </div>
  </Styled(MuiBox)>
</VideoSnippets>
`;
